osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs

```diff
@@ -383,21 +383,24 @@ private void handleDifficulty(string line)
             switch (pair.Key)
             {
                 case @"HPDrainRate":
-                    difficulty.DrainRate = Parsing.ParseFloat(pair.Value);
+                    difficulty.DrainRate = Math.Clamp(Parsing.ParseFloat(pair.Value), 0, 10);
                     break;
 
                 case @"CircleSize":
                     difficulty.CircleSize = Parsing.ParseFloat(pair.Value);
+                    //If the mode is not Mania, clamp circle size to [0,10]
+                    if (!beatmap.BeatmapInfo.Ruleset.OnlineID.Equals(3))
```

Comment:
Hmm... I did think of this, but sample I took had it being fine... But i suppose even if my sample was fine, that doesn't mean the user ain't gonna do something funky with editing the file.

I just wanted to avoid having to do checks in `LegacyBeatmapDecoder.ParseStreamInto`, originally my plan was just to change all the `BeatmapDifficulty` getters to do this, but it lead to way more complications then it was worth. (breaking stuff with realms, plus certain mods allow the user to edit difficulty settings up to 11, and the fact that you can't access what ruleset is being used from this class which required reworking how Mania checks circle size)

Anywho, latest commit should address this. Also added a clamp for when the mode is Mania - not as needed I suppose, but I think is good for consistency.